satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -897,24 +921,30 @@ def test_get_segment_position_info(self, reader_configs, fh_param, expected_pos_
             segpos_info = filetype_handler.get_segment_position_info()
             assert segpos_info == expected_pos_info
 
-    @mock.patch("satpy.readers.yaml_reader.GEOVariableSegmentYAMLReader")
     @pytest.mark.parametrize(("channel", "resolution"), generate_parameters("radiance"))
-    def test_not_get_segment_info_called_af(self,FakeFCIFileHandlerAF_fixture,reader_configs,channel,resolution):
+    def test_not_get_segment_info_called_af(self, FakeFCIFileHandlerAF_fixture, reader_configs, channel, resolution):
         """Test that checks that the get_segment_position_info has not been called for AF data."""
         with mock.patch("satpy.readers.fci_l1c_nc.FCIL1cNCFileHandler.get_segment_position_info") as gspi:
             fh_param = FakeFCIFileHandlerAF_fixture
             reader = _get_reader_with_filehandlers(fh_param["filenames"], reader_configs)
-            reader.load([channel])
+            try:
+                # attempt to load the channel
+                reader.load([channel])
+            except KeyError:
+                # If get_segment_position_info is called, the code will fail with a KeyError because of the mocking.
+                # So we catch the error here for now, but the test will still fail with the following assert_not_called
+                pass
```

Comment:
Is there something we can do besides except->pass? Maybe https://docs.python.org/3/library/contextlib.html#contextlib.suppress? Or what about no `except` at all? I guess maybe I'm confused why a KeyError would get to this level of execution and we shouldn't care about it?

Reply:
I switched to suppress, that's a good idea, thanks! I also tried to clarify a bit more my comment in the code for why the test is setup like this. 

In other words: without the fix of this PR, `get_segment_position_info` would be called (which we don't want), and would then currently fail with `KeyError`. My point is that we want to test that the function is not called at all, which is what the test is doing. If instead we would check for the `KeyError` to happen, someone could technically change `get_segment_position_info` to avoid the `KeyError`and not fail anymore, the test would then pass, but the bug/unexpected behaviour would likely still be there.